src/components/views/dialogs/BaseDialog.js

```diff
@@ -76,17 +72,22 @@ export default React.createClass({
         const TintableSvg = sdk.getComponent("elements.TintableSvg");
 
         return (
-            <div onKeyDown={this._onKeyDown} className={this.props.className}>
+            <FocusTrap onKeyDown={this._onKeyDown}
+                className={this.props.className}
+                role="dialog"
+                aria-labelledby='mx_BaseDialog_title'
+                aria-describedby={this.props.contentId}
```

Comment:
Yes by reading aria docs or best practices we can learn this should be pointing to a DOM node describing the dialog i.e. for simple yes/no dialog it is its whole content excluding the buttons. For more complex dialogs it would be some other node nested inside the dialog. If we were using it preciselly the way as suggested we will have to delegate setup of this prop to the component where the description is added. Thus we have opted for a compromise by using the whole dialog content as its accessible description. All of Riot dialogs have the descriptive text at the top thus usefull text would be presented by the assistive tools first
. Note AT users don't have to listen to the whole description, they can interupt it naturally whenever they are confortable to continue working with the app.

Reply:
OK. When making compromises like this, we generally comment them so we know what compromises have been made and why, otherwise everyone reading the code will end up asking the same question I asked.